src/model/SplineSegment.cpp

```diff
@@ -0,0 +1,83 @@
+#include "SplineSegment.h"
+
+#include <cmath>
+
+SplineSegment::SplineSegment(const Point& p, const Point& q) {
+    firstKnot = p;
+    firstControlPoint = p;
+    secondKnot = q;
+    secondControlPoint = q;
+}
+
+SplineSegment::SplineSegment(const Point& p, const Point& fp, const Point& sp, const Point& q) {
```

Comment:
This suggested change gives me a segfault error upon execution. The error comes from invoking `SplineSegment firstPart = SplineSegment(firstKnot, b0, c0, d0);` in the `SplineSegment::subdivide`-method.

Reply:
The member initialization list should be equivalent to the assignments. Did you copy the above change verbatim? This constructor assigns `fp` and `sp` as well.

If the member initialization list syntax doesn't work, it's fine to just forget it.

Reply:
The constructor that does not work (although it compiles fine) is
```
SplineSegment::SplineSegment(const` Point& p, const Point& fp, const Point& sp, const Point& q): 
        firstKnot(p), firstControlPoint(fp), secondKnot(sp), secondControlPoint(q) {}

```
It does assign `fp` and `sp` as well. So I will leave it as it is.

Reply:
I also tried it, and for me it works. There might be another problem. Do you tried to recompile the complete project?
```
SplineSegment::SplineSegment(const Point& p, const Point& fp, const Point& sp, const Point& q):
        firstKnot(p), secondKnot(q), firstControlPoint(fp), secondControlPoint(sp) {}
```

Reply:
Yes, I did. The program still crashes once I set the second knot (when the first tangent is non-trivial).

Reply:
The thing is, Technius and my code sniped must not result in a segfault. Your version and our's also produce exactly the same binary output (with optimizations enabled). The only case it can crash there is, when one of the Point references are deleted. Since 2 of them are members of the SplineSegment, the SplineSegment may be deleted. And that would be the real Problem. I'll try to recompile it with sanitizers on and look for the real cause.

Reply:
Ok I found the problem: It's a combination of using recursion, many growing vectors in a recursive function and the move into the return vector which may eliminate return value optimization. I'll list my change wish the following review comments.